lib/praxis-blueprints/field_expander.rb

```diff
@@ -26,7 +18,10 @@ def initialize
 
     def expand(object, fields=true)
       if stack[object].include? fields
-        raise CircularExpansionError, "Circular expansion detected for object #{object.inspect} with fields #{fields.inspect}"
+        if history[object].include? fields
+          return history[object][fields]
+        end
+        raise "Circular expansion detected for object #{object.inspect} with fields #{fields.inspect}"
```

Comment:
I would add a NOTE comment in there, that says something like: 
"it seems that if the stack has the object with fields, the history would need to have it too, so this raise would never every happen...but we're being conservative about catching it in case we're somehow missing some possible path"